### PR TITLE
[office-js, office-js-preview, office-runtime] change sso status from…

### PR DESCRIPTION
**Pull request by Rick-Kirkham**
… preview to release

Please fill in this template.

- [x ] Use a meaningful title for the pull request. Include the name of the package modified.
- [ ] Test the change in your own code. (Compile and run.)
- [ ] Add or edit tests to reflect the change. (Run with `npm test`.)
- [ x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x ] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [ ] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If changing an existing definition:
- [ ] Provide a URL to documentation or source code which provides context for the suggested changes: https://developer.microsoft.com/en-us/graph/blogs/announcing-general-availability-of-single-sign-on-sso-for-office-add-ins/
- [ ] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.
- [ ] Include [tests for your changes](https://github.com/DefinitelyTyped/DefinitelyTyped#testing)
- [ ] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`. If for reason the any rule need to be disabled, disable it for that line using `// tslint:disable-next-line [ruleName]` and not for whole package so that the need for disabling can be reviewed.

### CODE REVIEW

File: types/office-js/index.d.ts [hunk 1]
```diff
@@ -1021,7 +1021,100 @@ declare namespace Office {
...
+         *
+         * Important: In Outlook, this API is not supported if the add-in is loaded in an Outlook.com or Gmail mailbox.
+         *
+         * @remarks
+         *
+         * **Hosts**: Excel, OneNote, Outlook, PowerPoint, Word
+         *
+         * **Requirement set**: {@link https://docs.microsoft.com/office/dev/add-ins/reference/requirement-sets/identity-api-requirement-sets | IdentityAPI}
+         *
+         * This API requires a single sign-on configuration that bridges the add-in to an Azure application. Office users sign-in with Organizational
```
Comment (ElizabethSamuel-MSFT) [inline types/office-js/index.d.ts:1042]:
```suggestion
         * This API requires a single sign-on configuration that bridges the add-in to an Azure application. Office users sign in with Organizational
```

File: types/office-js/index.d.ts [hunk 2]
```diff
@@ -1021,7 +1021,100 @@ declare namespace Office {
...
+         * Accounts and Microsoft Accounts. Microsoft Azure returns tokens intended for both user account types to access resources in the Microsoft Graph.
+         *
+         * @param options - Optional. Accepts an AuthOptions object to define sign-on behaviors.
+         * @param callback - Optional. Accepts a callback method that can use parse the token for the user's ID or use the token in the "on behalf of" flow to get access to Microsoft Graph.
+         *                   If AsyncResult.status is "succeeded", then AsyncResult.value is the raw AAD v. 2.0-formatted access token.
+         */
+        getAccessTokenAsync(options?: AuthOptions, callback?: (result: AsyncResult<string>) => void): void;
+        /**
+         * Calls the Azure Active Directory V 2.0 endpoint to get an access token to your add-in's web application. Enables  add-ins to identify users.
+         * Server side code can use this token to access Microsoft Graph for the add-in's web application by using the
```
Comment (ElizabethSamuel-MSFT) [inline types/office-js/index.d.ts:1052]:
```suggestion
         * Server-side code can use this token to access Microsoft Graph for the add-in's web application by using the
```

File: types/office-js/index.d.ts [hunk 3]
```diff
@@ -1021,7 +1021,100 @@ declare namespace Office {
...
+         * {@link https://docs.microsoft.com/azure/active-directory/develop/active-directory-v2-protocols-oauth-on-behalf-of | "on behalf of" OAuth flow}.
+         *
+         * Important: In Outlook, this API is not supported if the add-in is loaded in an Outlook.com or Gmail mailbox.
```
Comment (ElizabethSamuel-MSFT) [inline types/office-js/index.d.ts:1055]:
```suggestion
         * **Important**: In Outlook, this API is not supported if the add-in is loaded in an Outlook.com or Gmail mailbox.
```

File: types/office-js/index.d.ts [hunk 4]
```diff
@@ -1021,7 +1021,100 @@ declare namespace Office {
...
+         *
+         * @remarks
+         *
+         * **Hosts**: Excel, OneNote, Outlook, PowerPoint, Word
+         *
+         * **Requirement set**: {@link https://docs.microsoft.com/office/dev/add-ins/reference/requirement-sets/identity-api-requirement-sets | IdentityAPI}
+         *
+         * This API requires a single sign-on configuration that bridges the add-in to an Azure application. Office users sign-in with Organizational
```
Comment (ElizabethSamuel-MSFT) [inline types/office-js/index.d.ts:1063]:
```suggestion
         * This API requires a single sign-on configuration that bridges the add-in to an Azure application. Office users sign in with Organizational
```

File: types/office-js/index.d.ts [hunk 5]
```diff
@@ -1021,7 +1021,100 @@ declare namespace Office {
...
+         * Accounts and Microsoft Accounts. Microsoft Azure returns tokens intended for both user account types to access resources in the Microsoft Graph.
+         *
+         * @param callback - Optional. Accepts a callback method that can use parse the token for the user's ID or use the token in the "on behalf of" flow to get access to Microsoft Graph.
+         *                   If AsyncResult.status is "succeeded", then AsyncResult.value is the raw AAD v. 2.0-formatted access token.
```
Comment (ElizabethSamuel-MSFT) [inline types/office-js/index.d.ts:1067]:
```suggestion
         *                   If `AsyncResult.status` is "succeeded", then `AsyncResult.value` is the raw AAD v. 2.0-formatted access token.
```

File: types/office-js/index.d.ts [hunk 6]
```diff
@@ -1021,7 +1021,100 @@ declare namespace Office {
...
+         */
+        getAccessTokenAsync(callback?: (result: AsyncResult<string>) => void): void;
+    }
+    /**
+     * Provides options for the user experience when Office obtains an access token to the add-in from AAD v. 2.0 with the getAccessTokenAsync method.
+     */
+    interface AuthOptions {
+        /**
+         * Allows Office to get an access token silently or through interactive consent, if one is required.
```
Comment (ElizabethSamuel-MSFT) [inline types/office-js/index.d.ts:1076]:
```suggestion
         * Allows Office to get an access token silently or through interactive consent, if one is required. Default value is `false`.
```

File: types/office-js/index.d.ts [hunk 7]
```diff
@@ -1021,7 +1021,100 @@ declare namespace Office {
...
+         * If set to false, Office will silently try to get an access token. If it fails to do so, Office will return a descriptive error.
```
Comment (ElizabethSamuel-MSFT) [inline types/office-js/index.d.ts:1077]:
```suggestion
         * If set to `false`, Office will silently try to get an access token. If it fails to do so, Office will return a descriptive error.
```

File: types/office-js/index.d.ts [hunk 8]
```diff
@@ -1021,7 +1021,100 @@ declare namespace Office {
...
+         * If set to true, Office will show an interactive consent UI after it fails to silently get an access token.
```
Comment (ElizabethSamuel-MSFT) [inline types/office-js/index.d.ts:1078]:
```suggestion
         * If set to `true`, Office will show an interactive consent UI after it fails to silently get an access token.
```

File: types/office-js/index.d.ts [hunk 9]
```diff
@@ -1021,7 +1021,100 @@ declare namespace Office {
...
+         * The prompt will only allow consent to the AAD profile scope, not to any Microsoft Graph scopes.
+         */
+        allowConsentPrompt?: boolean;
+        /**
+         * Allows Office to get an access token silently provided consent is present or show interactive UI to sign in the user.
```
Comment (ElizabethSamuel-MSFT) [inline types/office-js/index.d.ts:1083]:
```suggestion
         * Allows Office to get an access token silently provided consent is present or show interactive UI to sign in the user. Default value is `false`.
```

File: types/office-js/index.d.ts [hunk 10]
```diff
@@ -1021,7 +1021,100 @@ declare namespace Office {
...
+         * If set to false, office will silently try to get an access token. If it fails to do so, Office will return a descriptive error.
```
Comment (ElizabethSamuel-MSFT) [inline types/office-js/index.d.ts:1084]:
```suggestion
         * If set to `false`, Office will silently try to get an access token. If it fails to do so, Office will return a descriptive error.
```

File: types/office-js/index.d.ts [hunk 11]
```diff
@@ -1021,7 +1021,100 @@ declare namespace Office {
...
+         * If set to true, Office will show an interactive sign-in UI after it fails to silently get an access token.
```
Comment (ElizabethSamuel-MSFT) [inline types/office-js/index.d.ts:1085]:
```suggestion
         * If set to `true`, Office will show an interactive sign-in UI after it fails to silently get an access token.
```

File: types/office-js/index.d.ts [hunk 12]
```diff
@@ -1021,7 +1021,100 @@ declare namespace Office {
...
+         */
+        allowSignInPrompt?: boolean;
+        /**
+         * Deprecated:
+         * Prompts the user to add their Office account (or to switch to it, if it is already added).
```
Comment (ElizabethSamuel-MSFT) [inline types/office-js/index.d.ts:1090]: "@deprecated" tag should note what has replaced this property. See [deprecated documentation](https://api-extractor.com/pages/tsdoc/tag_deprecated/).
```suggestion
         * Prompts the user to add their Office account (or to switch to it, if it is already added). Default value is `false`.
         *
         * @deprecated
```

File: types/office-js/index.d.ts [hunk 13]
```diff
@@ -1021,7 +1021,100 @@ declare namespace Office {
...
+         */
+        forceAddAccount?: boolean;
+        /**
+         * Deprecated:
+         * Causes Office to display the add-in consent experience. Useful if the add-in's Azure permissions have changed or if the user's consent has
+         * been revoked.
```
Comment (ElizabethSamuel-MSFT) [inline types/office-js/index.d.ts:1096]: "@deprecated" tag should note what has replaced this property. See [deprecated documentation](https://api-extractor.com/pages/tsdoc/tag_deprecated/).
```suggestion
         * Causes Office to display the add-in consent experience. Useful if the add-in's Azure permissions have changed or if the user's consent has
         * been revoked. Default value is `false`.
         *
         * @deprecated
```

File: types/office-js/index.d.ts [hunk 14]
```diff
@@ -1021,7 +1021,100 @@ declare namespace Office {
...
+         */
+        forceConsent?: boolean;
+        /**
+         * Causes Office to prompt the user to provide the additional factor when the tenancy being targeted by Microsoft Graph requires multifactor
+         * authentication. The string value identifies the type of additional factor that is required. In most cases, you won't know at development
+         * time whether the user's tenant requires an additional factor or what the string should be. So this option would be used in a "second try"
+         * call of getAccessToken after Microsoft Graph has sent an error requesting the additional factor and containing the string that should
```
Comment (ElizabethSamuel-MSFT) [inline types/office-js/index.d.ts:1103]:
```suggestion
         * call of `getAccessTokenAsync` after Microsoft Graph has sent an error requesting the additional factor and containing the string that should
```

File: types/office-js/index.d.ts [hunk 15]
```diff
@@ -1021,7 +1021,100 @@ declare namespace Office {
...
+         * be used with the authChallenge option.
```
Comment (ElizabethSamuel-MSFT) [inline types/office-js/index.d.ts:1104]:
```suggestion
         * be used with the `authChallenge` option.
```

File: types/office-js/index.d.ts [hunk 16]
```diff
@@ -1021,7 +1021,100 @@ declare namespace Office {
...
+         */
+        authChallenge?: string;
+        /**
+         * A user-defined item of any type that is returned, unchanged, in the asyncContext property of the AsyncResult object that is passed to a callback.
```
Comment (ElizabethSamuel-MSFT) [inline types/office-js/index.d.ts:1108]:
```suggestion
         * A user-defined item of any type that is returned, unchanged, in the `asyncContext` property of the `AsyncResult` object that is passed to a callback.
```

File: types/office-js/index.d.ts [hunk 17]
```diff
@@ -1021,7 +1021,100 @@ declare namespace Office {
...
+         */
+        asyncContext?: any;
+        /**
+         * Causes Office to return descriptive error when the add-in wants to access MS Graph and the user/admin has not granted consent to MS Graph scopes.
```
Comment (ElizabethSamuel-MSFT) [inline types/office-js/index.d.ts:1112]:
```suggestion
         * Causes Office to return a descriptive error when the add-in wants to access Microsoft Graph and the user/admin has not granted consent to Graph scopes.
```

File: types/office-js/index.d.ts [hunk 18]
```diff
@@ -1021,7 +1021,100 @@ declare namespace Office {
...
+         * Office only supports consent to graph scopes when the add-in has been deployed by a tenant admin. This information will not be available during development.
```
Comment (ElizabethSamuel-MSFT) [inline types/office-js/index.d.ts:1113]:
```suggestion
         * Office only supports consent to Graph scopes when the add-in has been deployed by a tenant admin. This information will not be available during development.
```

File: types/office-js/index.d.ts [hunk 19]
```diff
@@ -1021,7 +1021,100 @@ declare namespace Office {
...
+         * Setting this option to true will allow Office to inform your add-in beforehand if MS graph access will fail by returning back a descriptive error.
```
Comment (ElizabethSamuel-MSFT) [inline types/office-js/index.d.ts:1114]:
```suggestion
         * Setting this option to `true` will allow Office to inform your add-in beforehand by returning a descriptive error if Graph access will fail.
         * Default value is `false`.
```

File: types/office-js/index.d.ts [hunk 20]
```diff
@@ -1021,7 +1021,100 @@ declare namespace Office {
...
+         * @remarks
+         *
+         * **Hosts**: Excel, OneNote, Outlook, PowerPoint, Word
+         *
+         * **Requirement set**: {@link https://docs.microsoft.com/office/dev/add-ins/reference/requirement-sets/identity-api-requirement-sets | IdentityAPI}
+         *
+         * This API requires a single sign-on configuration that bridges the add-in to an Azure application. Office users sign-in with Organizational
+         * Accounts and Microsoft Accounts. Microsoft Azure returns tokens intended for both user account types to access resources in the Microsoft Graph.
+         *
+         * @param callback - Optional. Accepts a callback method that can use parse the token for the user's ID or use the token in the "on behalf of" flow to get access to Microsoft Graph.
```
Comment (ElizabethSamuel-MSFT) [inline types/office-js/index.d.ts:1066]: "use" or "parse"
```suggestion
         * @param callback - Optional. Accepts a callback method that can use the token for the user's ID or use the token in the "on behalf of" flow to get access to Microsoft Graph.
```

File: types/office-runtime/index.d.ts [hunk 21]
```diff
@@ -226,9 +228,7 @@ declare namespace OfficeRuntime {
      *
      * @remarks
      *
-     * **Hosts**: Excel, PowerPoint, Word
-     *
-     * @beta
+     * **Hosts**: Excel, PowerPoint, Word, Outlook
```
Comment (ElizabethSamuel-MSFT) [inline types/office-runtime/index.d.ts:231]:
```suggestion
     * **Hosts**: Excel, Outlook, PowerPoint, Word
```

**Review by ElizabethSamuel-MSFT — CHANGES_REQUESTED** [review]
If you accept any suggestions, I would recommend making all of them in office-js then copying the entire Auth and AuthOptions objects to office-js-preview and office-runtime to ensure consistency.

File: types/office-js-preview/index.d.ts [hunk 22]
```diff
@@ -1181,99 +1181,101 @@ declare namespace Office {
...
          * time whether the user's tenant requires an additional factor or what the string should be. So this option would be used in a "second try"
-         * call of getAccessToken after Microsoft Graph has sent an error requesting the additional factor and containing the string that should
-         * be used with the authChallenge option.
+         * call of `getAccessTokenAsync` after Microsoft Graph has sent an error requesting the additional factor and containing the string that should
+         * be used with the `authChallenge` option.
          */
         authChallenge?: string;
         /**
-         * A user-defined item of any type that is returned, unchanged, in the asyncContext property of the AsyncResult object that is passed to a callback.
+         * A user-defined item of any type that is returned, unchanged, in the `asyncContext property` of the `AsyncResult` object that is passed to a callback.
```
Comment (ElizabethSamuel-MSFT) [inline types/office-js-preview/index.d.ts:1269]:
```suggestion
         * A user-defined item of any type that is returned, unchanged, in the `asyncContext` property of the `AsyncResult` object that is passed to a callback.
```

File: types/office-js/index.d.ts [hunk 23]
```diff
@@ -1022,96 +1022,98 @@ declare namespace Office {
         asyncContext?: any
     }
     /**
-     * The Office Auth namespace, Office.context.auth, provides a method that allows the Office host to obtain an access token to the add-in's web application.
+     * The Office Auth namespace, Office.context.auth, provides a method that allows the Office client application to obtain an access token to the add-in's web application.
```
Comment (ElizabethSamuel-MSFT) [inline types/office-js/index.d.ts:1025]:
```suggestion
     * The Office Auth namespace, `Office.context.auth`, provides a method that allows the Office client application to obtain an access token to the add-in's web application.
```

File: types/office-js/index.d.ts [hunk 24]
```diff
@@ -1022,96 +1022,98 @@ declare namespace Office {
...
          * time whether the user's tenant requires an additional factor or what the string should be. So this option would be used in a "second try"
-         * call of getAccessToken after Microsoft Graph has sent an error requesting the additional factor and containing the string that should
-         * be used with the authChallenge option.
+         * call of `getAccessTokenAsync` after Microsoft Graph has sent an error requesting the additional factor and containing the string that should
+         * be used with the `authChallenge` option.
          */
         authChallenge?: string;
         /**
-         * A user-defined item of any type that is returned, unchanged, in the asyncContext property of the AsyncResult object that is passed to a callback.
+         * A user-defined item of any type that is returned, unchanged, in the `asyncContext property` of the `AsyncResult` object that is passed to a callback.
```
Comment (ElizabethSamuel-MSFT) [inline types/office-js/index.d.ts:1110]:
```suggestion
         * A user-defined item of any type that is returned, unchanged, in the `asyncContext` property of the `AsyncResult` object that is passed to a callback.
```

File: types/office-runtime/index.d.ts [hunk 25]
```diff
@@ -166,71 +166,73 @@ declare namespace OfficeRuntime {
...
-     * A user-defined item of any type that is returned, unchanged, in the asyncContext property of the AsyncResult object that is passed to a callback.
-     */
-    asyncContext?: any;
-    /**
-     * Causes Office to return descriptive error when the add-in wants to access MS Graph and the user/admin has not granted consent to MS Graph scopes.
-     * Office only supports consent to graph scopes when the add-in has been deployed by a tenant admin. This information will not be available during development.
-     * Setting this option to true will allow Office to inform your add-in beforehand if MS graph access will fail by returning back a descriptive error.
-     */
-    forMSGraphAccess?: boolean;
+     * Provides options for the user experience when Office obtains an access token to the add-in from AAD v. 2.0 with the `getAccessToken` method.
```
Comment (ElizabethSamuel-MSFT) [inline types/office-runtime/index.d.ts:170]:
```suggestion
     * Provides options for the user experience when Office obtains an access token to the add-in from AAD v. 2.0 with the `getAccessTokenAsync` method.
```

File: types/office-runtime/index.d.ts [hunk 26]
```diff
@@ -166,71 +166,73 @@ declare namespace OfficeRuntime {
...
+       * been revoked. Default value is `false`.
+       *
+       * @deprecated Use `allowConsentPrompt` instead.
+       */
+      forceConsent?: boolean;
+      /**
+       * Causes Office to prompt the user to provide the additional factor when the tenancy being targeted by Microsoft Graph requires multifactor
+       * authentication. The string value identifies the type of additional factor that is required. In most cases, you won't know at development
+       * time whether the user's tenant requires an additional factor or what the string should be. So this option would be used in a "second try"
+       * call of `getAccessToken` after Microsoft Graph has sent an error requesting the additional factor and containing the string that should
```
Comment (ElizabethSamuel-MSFT) [inline types/office-runtime/index.d.ts:203]:
```suggestion
       * call of `getAccessTokenAsync` after Microsoft Graph has sent an error requesting the additional factor and containing the string that should
```

File: types/office-runtime/index.d.ts [hunk 27]
```diff
@@ -166,71 +166,73 @@ declare namespace OfficeRuntime {
...
+       * be used with the `authChallenge` option.
+       */
+      authChallenge?: string;
+      /**
+       * A user-defined item of any type that is returned, unchanged, in the `asyncContext property` of the `AsyncResult` object that is passed to a callback.
```
Comment (ElizabethSamuel-MSFT) [inline types/office-runtime/index.d.ts:208]:
```suggestion
       * A user-defined item of any type that is returned, unchanged, in the `asyncContext` property of the `AsyncResult` object that is passed to a callback.
```

File: types/office-runtime/index.d.ts [hunk 28]
```diff
@@ -166,71 +166,73 @@ declare namespace OfficeRuntime {
...
-     * Important: In Outlook, this API is not supported if the add-in is loaded in an Outlook.com or Gmail mailbox.
+     * **Important**: In Outlook, this API is not supported if the add-in is loaded in an Outlook.com or Gmail mailbox.
      *
      * @remarks
      *
-     * **Hosts**: Excel, PowerPoint, Word, Outlook
+     * **Hosts**: Excel, Outlook, PowerPoint, Word
      *
-     * @param options - Optional. Accepts an AuthOptions object to define sign-on behaviors.
+     * @param options - Optional. Accepts an `AuthOption`s object to define sign-on behaviors.
```
Comment (ElizabethSamuel-MSFT) [inline types/office-runtime/index.d.ts:235]:
```suggestion
     * @param options - Optional. Accepts an `AuthOptions` object to define sign-on behaviors.
```

File: types/office-runtime/index.d.ts (hunk 25 above)
Comment (Rick-Kirkham) [reply types/office-runtime/index.d.ts:170]: @ElizabethSamuel-MSFT In the OfficeRuntime namespace, the method is getAccessToken, not getAccessTokenAsync as it is in the Office namespace.

Comment (ElizabethSamuel-MSFT) [reply types/office-runtime/index.d.ts:170]: Ah, I see. Thanks.

File: types/office-js-preview/index.d.ts [hunk 29]
```diff
@@ -1181,83 +1181,100 @@ declare namespace Office {
     }
 
     /**
-     * The Office Auth namespace, Office.context.auth, provides a method that allows the Office host to obtain an access token to the add-in's web application.
+     * The Office Auth namespace, Office.context.auth, provides a method that allows the Office client application to obtain an access token to the add-in's web application.
```
Comment (ElizabethSamuel-MSFT) [inline types/office-js-preview/index.d.ts:1184]:
```suggestion
     * The Office Auth namespace,` Office.context.auth`, provides a method that allows the Office client application to obtain an access token to the add-in's web application.
```

**Comment by danger-public** [comment]
Inspecting the JavaScript source for this package found some properties that are not in the .d.ts files.
The check for missing properties isn't always right, so take this list as advice, not a requirement.
## bulma-calendar ([<kbd>unpkg</kbd>](https://unpkg.com/browse/bulma-calendar@latest/))
was missing the following properties:
1. bulmaCalendar
<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against 67f0e1cde4a9426026cb587944f728e4dc7ea6a0
</p>

**Review by ElizabethSamuel-MSFT — APPROVED** [review]
One final update. Looks good.

**Pull request merged**